src/Text/Parsing/Parser/Combinators.purs

```diff
@@ -99,30 +100,33 @@ lookAhead p = (ParserT <<< ExceptT <<< StateT) \s -> do
 -- | digit `sepBy` string ","
 -- | ```
 sepBy :: forall m s a sep. Monad m => ParserT s m a -> ParserT s m sep -> ParserT s m (List a)
-sepBy p sep = sepBy1 p sep <|> pure Nil
+sepBy p sep = (toList <$> sepBy1 p sep) <|> pure Nil
 
 -- | Parse phrases delimited by a separator, requiring at least one match.
-sepBy1 :: forall m s a sep. Monad m => ParserT s m a -> ParserT s m sep -> ParserT s m (List a)
+sepBy1 :: forall m s a sep. Monad m => ParserT s m a -> ParserT s m sep -> ParserT s m (NonEmptyList a)
 sepBy1 p sep = do
   a <- p
   as <- many $ sep *> p
-  pure (a : as)
+  pure (cons' a as)
 
 -- | Parse phrases delimited and optionally terminated by a separator.
 sepEndBy :: forall m s a sep. Monad m => ParserT s m a -> ParserT s m sep -> ParserT s m (List a)
-sepEndBy p sep = sepEndBy1 p sep <|> pure Nil
+sepEndBy p sep = (toList <$> sepEndBy1 p sep) <|> pure Nil
```

Comment:
Would it be beneficial to separate these related combinators, so the non-empty combinators don't depend on their possibly-empty counterparts and vice-versa?

Reply:
Yes, I think that’s sensible.

Reply:
Updated.